cmd/otel-allocator/README.md

```diff
@@ -122,6 +123,51 @@ In essence, Prometheus Receiver configs are overridden with a `http_sd_config` d
 Allocator, these are then loadbalanced/sharded to the Collectors. The [Prometheus Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/README.md) configs that are overridden
 are what will be distributed with the same name.
 
+## TargetAllocator CRD
```

Comment:
Nit: how about documenting the `operator.collector.targetallocatorcr` feature flag and explain what is the effect regarding this CRD?

Reply:
I'm not sure that's helpful at this point. That flag controls how the collector CR is reconciled, it doesn't have anything to do with the TargetAllocator CR per se. I introduced it specifically so that this PR doesn't change any existing code paths.